A way to make SourceChecker.suppressWarningsString return _only_ the prefix

[In the old JSpecify fork](https://github.com/jspecify/checker-framework/blob/b84882d9b2f77c99081118cc7ad095dfcc3484f5/framework/src/main/java/org/checkerframework/framework/source/SourceChecker.java#L1323-L1325), we had code that let us produce errors like:

```
com/google/common/collect/TreeMultimap.java:222: error: [nullness] incompatible types in assignment.
    keyComparator = (Comparator<? super K>) stream.readObject();
                    ^
  found   : Comparator<? super K>?
  required: Comparator<? super K>
```

Without that support, our errors now look like this:

```
com/google/common/collect/TreeMultimap.java:222: error: [assignment.type.incompatible] incompatible types in assignment.
```

It appears that we could enable `-ArequirePrefixInWarningSuppressions` or `-AshowPrefixInWarningMessages` to produce `[nullness:assignment.type.incompatible]`, which at least gets `nullness` into the picture. But we'd been experimenting with the shorter name (which necessarily means that suppressions are more blunt).

--- a/docs/CHANGELOG.md
+++ b/docs/CHANGELOG.md
@@ -8,6 +8,8 @@ Use `-ApermitMissingJdk` instead.
 
 **Implementation details:**
 
+Make `SourceChecker#suppressWarningsString` protected to allow adaptation in subclasses.
+
 **Closed issues:**
 
 

--- a/framework/src/main/java/org/checkerframework/framework/source/SourceChecker.java
+++ b/framework/src/main/java/org/checkerframework/framework/source/SourceChecker.java
@@ -1519,7 +1519,7 @@ public abstract class SourceChecker extends AbstractTypeProcessor implements Opt
      * @param messageKey the simple, checker-specific error message key
      * @return the most specific SuppressWarnings string for the warning/error being printed
      */
-    private String suppressWarningsString(String messageKey) {
+    protected String suppressWarningsString(String messageKey) {
         Collection<String> prefixes = this.getSuppressWarningsPrefixes();
         prefixes.remove(SUPPRESS_ALL_PREFIX);
         if (showSuppressWarningsStrings) {

